In [1]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_digits, load_breast_cancer
from sklearn.model_selection import cross_val_score
from sklearn.naive_bayes import MultinomialNB, GaussianNB, BernoulliNB

C:\ProgramData\Anaconda3\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 112 from C header, got 124 from PyObject
  return f(*args, **kwds)
C:\ProgramData\Anaconda3\lib\importlib\_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 112 from C header, got 124 from PyObject
  return f(*args, **kwds)


In [18]:
dig = load_digits()
X_dig = dig.data
y_dig = dig.target
#print(dig)

In [20]:
canc = load_breast_cancer()
X_canc = canc.data
y_canc = canc.target
#print(canc)

In [24]:
mnb = MultinomialNB()
gnb = GaussianNB()
bnb = BernoulliNB()

dig_acc_mnb = cross_val_score(mnb,X_dig,y_dig).mean()
dig_acc_gnb = cross_val_score(gnb,X_dig,y_dig).mean()
dig_acc_bnb = cross_val_score(bnb,X_dig,y_dig).mean()

canc_acc_mnb = cross_val_score(mnb,X_canc,y_canc).mean()
canc_acc_gnb = cross_val_score(gnb,X_canc,y_canc).mean()
canc_acc_bnb = cross_val_score(bnb,X_canc,y_canc).mean()

print('Accuracy(digits dataset) : \n mnb: {} \n gnb {} \n bnb {}  \n'.format(dig_acc_mnb,dig_acc_gnb,dig_acc_bnb))
print('Accuracy(breast cancer dataset) : \n mnb: {} \n gnb {} \n bnb {} '.format(canc_acc_mnb,canc_acc_gnb,canc_acc_bnb))

Accuracy(digits dataset) : 
 mnb: 0.8708771489735053 
 gnb 0.8186003803550138 
 bnb 0.8258236507780582  

Accuracy(breast cancer dataset) : 
 mnb: 0.8945790401930752 
 gnb 0.9367492806089297 
 bnb 0.6274204028589994 


C:\ProgramData\Anaconda3\lib\site-packages\sklearn\model_selection\_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.
  warnings.warn(CV_WARNING, FutureWarning)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\model_selection\_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.
  warnings.warn(CV_WARNING, FutureWarning)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\model_selection\_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.
  warnings.warn(CV_WARNING, FutureWarning)
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\model_selection\_split.py:1978: FutureWarning: The default value of cv will change from 3 to 5 in version 0.22. Specify it explicitly to silence this warning.
  warnings.warn(CV_WARNING, Fut

Вопрос 1

Каким получилось максимальное качество классификации на датасете breast_cancer?
gnb 0.9367492806089297
Вопрос 2

Каким получилось максимальное качество классификации на датасете digits?
mnb: 0.8708771489735053
Вопрос 3

Выберите верные утверждения и запишите их номера через пробел (в порядке возрастания номера):

~~1) На вещественных признаках лучше всего сработал наивный байесовский классификатор с распределением Бернулли~~

~~2) На вещественных признаках лучше всего сработал наивный байесовский классификатор с мультиномиальным распределением~~

3) Мультиномиальное распределение лучше показало себя на выборке с целыми неотрицательными значениями признаков

4) На вещественных признаках лучше всего сработало нормальное распределение
